packages/pigeon/lib/dart_generator.dart

```diff
@@ -332,16 +336,17 @@ $resultAt != null
                 'binaryMessenger: binaryMessenger);',
               );
             });
-            final String messageHandlerSetter =
-                isMockHandler ? 'setMockMessageHandler' : 'setMessageHandler';
+            final String messageHandlerSetter = isMockHandler
+                ? '_testBinaryMessengerBinding!.defaultBinaryMessenger.setMockDecodedMessageHandler<Object?>(channel, '
```

Comment:
is the `!` here the _ambiguate equivalent?

Reply:
Yes, that plus typing the return type of the getter above as `TestDefaultBinaryMessengerBinding?` to hide the fact that `TestDefaultBinaryMessengerBinding.instance` may be non-nullable in certain versions of Flutter.